spec/clearance/session_spec.rb

```diff
@@ -43,6 +43,55 @@
     after { restore_default_config }
   end
 
+  context "without a custom cookie name" do
+    it "does not specify the cookie domain" do
+      Clearance.configuration.cookie_domain = nil
+
+      session.sign_in user
+      session.add_cookie_to_headers(headers)
+
+      expect(headers["Set-Cookie"]).not_to match(/domain=/)
+    end
+  end
+
+  context "with multiple allowable domains" do
+    let(:session_stubbed){ session.dup }
+
+    before do
+      Clearance.configuration.cookie_domain = %w(example.com example.org)
+
+      session_stubbed.sign_in user
+    end
+
+    it "permits whitelisted hosts" do
+      allow(session_stubbed).to receive(:get_request_uri).and_return 'http://example.org/foo'
+      session_stubbed.add_cookie_to_headers(headers)
+
+      expect(headers["Set-Cookie"]).to match(/example.org/)
+    end
+
+    it "defaults to the first whitelisted host when no header is present" do
+      allow(session_stubbed).to receive(:get_request_uri).and_return 'http://127.0.0.1/foo'
+      session_stubbed.add_cookie_to_headers(headers)
+
+      expect(headers["Set-Cookie"]).to match(/example.com/)
+    end
+
+    it "defaults to the first whitelisted host when other host headers are present" do
+      allow(session_stubbed).to receive(:get_request_uri).and_return 'http://example.net/foo'
+      session_stubbed.add_cookie_to_headers(headers)
+
+      expect(headers["Set-Cookie"]).to match(/example.com/)
+    end
+
+    it "ignores urls that fail to parse for any reason" do
+      allow(session_stubbed).to receive(:get_request_uri).and_return 'invalid domain name'
```

Comment:
Line is too long. [90/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.